src/main/java/fi/aalto/cs/apluscourses/model/Group.java

```diff
@@ -0,0 +1,76 @@
+package fi.aalto.cs.apluscourses.model;
+
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import fi.aalto.cs.apluscourses.utils.CoursesClient;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+public class Group {
+
+  private long id;
+
+  private List<String> memberNames;
+
+  public Group(long id, @NotNull List<String> memberNames) {
+    this.id = id;
+    this.memberNames = memberNames;
+  }
+
+  /**
+   * Construct a group from the given JSON object. The JSON object must contain an integer ID for
+   * the key "id", as well as an array for the key "members", where each value of the array is a
+   * JSON object containing a string for the key "full_name".
+   */
+  @NotNull
+  public static Group fromJsonObject(@NotNull JSONObject jsonObject) {
+    long id = jsonObject.getLong("id");
+    JSONArray members = jsonObject.getJSONArray("members");
+    List<String> memberNames = new ArrayList<>(members.length());
+    for (int i = 0; i < members.length(); ++i) {
+      memberNames.add(members.getJSONObject(i).getString("full_name"));
+    }
+    return new Group(id, memberNames);
```

Comment:
Remark: There are quite a lot of these JSON parsing functions with very similar logics in our model classes. At some point, it would be good to do some refactoring and separate lower level parsing code from the domain object building.

Reply:
Good point! :+1: Although this parsing code is still on quite a high level of abstraction so it's not too bad for now